agent/api/task/task.go

```diff
@@ -334,14 +334,13 @@ func (task *Task) PostUnmarshalTask(cfg *config.Config,
 	if err != nil {
 		return apierrors.NewResourceInitError(task.Arn, err)
 	}
-	if cfg.GPUSupportEnabled {
-		err = task.addGPUResource()
-		if err != nil {
-			seelog.Errorf("Task [%s]: could not initialize GPU associations: %v", task.Arn, err)
-			return apierrors.NewResourceInitError(task.Arn, err)
-		}
-		task.NvidiaRuntime = cfg.NvidiaRuntime
+
+	err = task.addGPUResource(cfg)
```

Comment:
if you wanted to, you could condense:

```
if err := foo(); err != nil {
   return err
}

Reply:
I'll fix it in a separate PR, we need to create PRs to test event handling.